### PR TITLE
Avoid node selection if default is prevented

### PR DESCRIPTION
**Pull request by Llorx**
I use HTML elements (inputs, dropdowns, checkboxes and such) as cell values. When I click an element, `cellClickHandler` calls `click.jstree` event before `select_cell.jstree-grid` is called, making jstree call `$(e.currentTarget).focus();` to focus the anchor element and so losing the focus on the input element:

This is a problem because the dropdown or input focus is lost. I can regain the lost focus in the callback, but some things happen:
 - On `select` elements the only way to reopen it is by creating a fake `click` event. This will recall the `cellClickHandler` again, that will call my callback and again call the `click` event, and so on.
 - On text inputs, sometimes the selection or caret position is lost.
 - In short, is not optimal to lose focus and regain it when not needed.

To fix this I've added a check to avoid jstree node selection, if the user wants, by calling `event.preventDefault()`, and do the selection logic on its side if he wants (calling `select_node`, for example).

### CODE REVIEW

**Comment by deitch** [comment]
Took a little while to get back to this.

1. this is really good. It does solve a real issue. Thank you, indeed.
2. how does one invoke the `event.preventDefault()`? I see in the code `event.isDefaultPrevented()` (which, I assume, is why you created a separate `event`), but where does `event.preventDefault()` get called?
3. Why do we eliminate the `node.children(".jstree-anchor").trigger("click.jstree",e);`?
4. Can you add a few lines to the README indicating when you would use this and how it is invoked (which just goes back to 2 above)?
5. Can you update the test files to test for this? I know, they are all manual, but at least they are there. Just copy any file from within https://github.com/deitch/jstree-grid/tree/master/tests/components and modify to show your use case.

**Comment by Llorx** [comment]
1. Np :P
2. When you hear the `select_cell.jstree-grid` event, you receive an `event` argument that can be prevented, this way:
``` js
mytree.on("select_cell.jstree-grid", function(event, data) {
    if (/* some logic, if you want */) {
        event.preventDefault();
    }
});
```
3. Is not eliminated. Is called below just if the event is not prevented. Preventing the event blocks the `click.jstree` call, which is the one creating the problem. In other words, preventing the `select_cell.jstree-grid` will prevent the entire line selection, avoiding losing focus because of jstree internals when clicking an element. If you truly want to select a line after preventing the event but without losing focus, you can do inside the handler (as I do):

``` js
mytree.on("select_cell.jstree-grid", function(event, data) {
    event.preventDefault();
    mytree.jstree("deselect_all");
    mytree.jstree("select_node", data.node);
});
```
but that's up to the user. Maybe he just wants to block the line selection and keep the old selection or keep the selection blank.
4. Sure
5. Sure :P

**Comment by deitch** [comment]
> 2. When you hear the select_cell.jstree-grid event, you receive an event argument that can be prevented, this way:

That makes sense. 

> 3. Is not eliminated. Is called below just if the event is not prevented.

Ah, you just switched the order. I see. That makes a lot of sense.

**Comment by Llorx** [comment]
Ok, commit created. Feel free to use your own words to edit my text in the readme. English is not my main language the text I've written sounds weird for sure xD

**Comment by deitch** [comment]
It is pretty good actually. I am checking out now to check. (that sounds funny...)

**Comment by deitch** [comment]
I made some small changes to the test to make the cases clearer, and split the README a bit, but overall good. Thank you!
